pkg/services/apiserver/builder/helper.go

```diff
@@ -144,15 +150,34 @@ func InstallAPIs(
 	builders []APIGroupBuilder,
 	storageOpts *options.StorageOptions,
 	reg prometheus.Registerer,
+	kvStore grafanarest.NamespacedKVStore,
+	serverLock ServerLockService,
 ) error {
 	// dual writing is only enabled when the storage type is not legacy.
 	// this is needed to support setting a default RESTOptionsGetter for new APIs that don't
 	// support the legacy storage type.
 	dualWriteEnabled := storageOpts.StorageType != options.StorageTypeLegacy
 
 	for _, b := range builders {
-		mode := b.GetDesiredDualWriterMode(dualWriteEnabled, storageOpts.DualWriterDesiredModes)
-		g, err := b.GetAPIGroupInfo(scheme, codecs, optsGetter, mode, reg)
+		var dualWrite grafanarest.DualWriteBuilder
+		if dualWriteEnabled {
+			mode := storageOpts.DualWriterDesiredModes[b.GetGroupVersion().Group] // defaults to 0
```

Comment:
@ryantxu In production this map is using a concatenation of `Group` and `Resource` as key, not `Group` 
(for example `playlist.grafana.app/playlists` instead of `playlist.grafana.app`)

We would have to change this PR to be something like the snippet below, which would require to modify builders to expose the GroupResource information:

```suggestion
			mode := storageOpts.DualWriterDesiredModes[b.GetGroupResource().Group + "/" + b.GetGroupResource().Resource] // defaults to 0
```

Another option, would be to modify Hosted Grafana to inject the instance config using only GROUP as key.
(would require code changes to HG API)

I don't know why the format `group / resource` was chosen, so I would like to hear from @suntala and @leonorfmartins in order to get more context on this.

Reply:
The format was chosen as part of this PR discussion: https://github.com/grafana/grafana/pull/88353#discussion_r1617307121